clang/lib/Sema/SemaOverload.cpp

```diff
@@ -10088,9 +10088,13 @@ static bool allowAmbiguity(ASTContext &Context, const FunctionDecl *F1,
                            const FunctionDecl *F2) {
   if (declaresSameEntity(F1, F2))
     return true;
-  if (F1->isTemplateInstantiation() && F2->isTemplateInstantiation() &&
-      declaresSameEntity(F1->getPrimaryTemplate(), F2->getPrimaryTemplate())) {
-    return true;
+  if (F1->isTemplateInstantiation() && F2->isTemplateInstantiation()) {
+    auto PT1 = F1->getPrimaryTemplate();
+    auto PT2 = F2->getPrimaryTemplate();
+    if (declaresSameEntity(PT1, PT2) ||
```

Comment:
Could we remove the check `getPrimaryTemplate` and instead only look at `getInstantiatedFromMemberTemplate`?

If `getPrimaryTemplate` matches, the `getInstantiatedFromMemberTemplate` will always match as well, right?

Reply:
Also, I believe the crashes [here](https://buildkite.com/llvm-project/clang-ci/builds/6710#018bcc9e-fdcb-44f8-802e-f5e98303916c) are related: `getCommonPtr()` that you see in the stack trace is called  by `getInstantiatedFromMemberTemplate()`.

Reply:
Added test